storm-client/src/jvm/org/apache/storm/task/TopologyContext.java

```diff
@@ -57,13 +57,14 @@
     private Map<String, Object> _executorData;
     private Map<Integer,Map<Integer, Map<String, IMetric>>> _registeredMetrics;
     private AtomicBoolean _openOrPrepareWasCalled;
-
+    private Map<String, Long> _blobToLastKnownVersion;
 
     public TopologyContext(StormTopology topology,
                            Map<String, Object> topoConf,
                            Map<Integer, String> taskToComponent,
                            Map<String, List<Integer>> componentToSortedTasks,
                            Map<String, Map<String, Fields>> componentToStreamToFields,
+                           Map<String, Long> blobToLastKnownVersion,
```

Comment:
So this is a little odd in that we send the versions to all of the children by passing in a variable that we expect to change behind the scenes.  I think it would be good to call that out in some comments both in the constructor here, and on field too so it is obvious here in the code what is happening.